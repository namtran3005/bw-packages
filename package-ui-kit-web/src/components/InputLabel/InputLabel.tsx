import * as React from 'react';
import { Typography, TypographyProps } from '../Typography';

interface InputLabelProps extends TypographyProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const InputLabel: React.FunctionComponent<InputLabelProps> = ({
  foreground = 'text',
  ...props
}) => <Typography.Small foreground={foreground} {...props} />;

export { InputLabel, InputLabelProps };
