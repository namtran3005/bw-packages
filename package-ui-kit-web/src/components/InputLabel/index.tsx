export { InputLabel, InputLabelProps } from './InputLabel';
