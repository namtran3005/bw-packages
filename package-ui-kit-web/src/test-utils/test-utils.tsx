import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
