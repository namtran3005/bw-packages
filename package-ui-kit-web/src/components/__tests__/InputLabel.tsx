import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { InputLabel } from '../InputLabel';

const defaultProps = {};

describe('InputLabel', () => {
  it('should render input label', () => {
    const { baseElement } = renderWithTestHarness(
      <InputLabel {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
