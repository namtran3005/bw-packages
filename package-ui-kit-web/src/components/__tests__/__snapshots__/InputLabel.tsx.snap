// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel should render input label 1`] = `
.c0 {
  text-align: left;
}

.c0.c0.c0 {
  font-weight: 400;
  font-size: 0.75rem;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c0.c0.c0:hover {
  background: initial;
}

.c0.c0.c0:active {
  background: initial;
}

<body>
  <div>
    <p
      class="MuiTypography-root c0 MuiTypography-body2"
    />
  </div>
</body>
`;
